### PR TITLE
correctness fixes to get Envoy xDS protos compiling

### PR DESCRIPTION
**Pull request by bpowers**
👋🏻 I'm working to read in some Envoy protobufs, and its a large project with a bunch of corner cases (e.g. fields named `map`, enum values named `self` and with the `allow_alias` option enabled, etc).   The below commits are what I needed to get things at least compiling!

### CODE REVIEW

File: pbjson-build/src/escape.rs [hunk 1]
```diff
@@ -24,3 +24,12 @@ pub fn escape_ident(mut ident: String) -> String {
     };
     ident
 }
+
+/// Converts a `snake_case` identifier to an `UpperCamel` case Rust type identifier.
+pub fn escape_camel_case(mut ident: String) -> String {
```
Comment (tustvold) [inline pbjson-build/src/escape.rs:29]:
```suggestion
pub fn escape_type(mut ident: String) -> String {
```

File: pbjson-build/src/resolver.rs [hunk 2]
```diff
@@ -57,7 +58,7 @@ impl<'a> Resolver<'a> {
                     .path()
                     .iter()
                     .zip(path.path())
-                    .filter(|(a, b)| a == b)
+                    .take_while(|(a, b)| a == b)
```
Comment (tustvold) [inline pbjson-build/src/resolver.rs:61]: Nice :ok_hand:

**Review by tustvold — COMMENTED** [review]
I am so sorry for missing this and not reviewing this sooner, thank you for this contribution :+1: 

I left a minor nit, which you can take or leave, just let me know.

If you could please sign the [CLA](https://www.influxdata.com/legal/cla/), I can get this in. It's pretty boilerplate, but let me know if there are any issues.

File: pbjson-build/src/generator/enumeration.rs [hunk 3]
```diff
@@ -21,9 +23,16 @@ pub fn generate_enum<W: Write>(
...
 
+    let mut numbers = HashSet::new();
+
     let variants: Vec<_> = descriptor
         .values
         .iter()
+        .filter(|variant| {
+            // Skip duplicate enum values. Protobuf allows this when the
+            // 'allow_alias' option is set.
+            numbers.insert(variant.number())
```
Comment (tustvold) [inline pbjson-build/src/generator/enumeration.rs:34]: Perhaps we could get a test of this added to pbjson-test?

**Comment by kaizensparc** [comment]
Hi! Is there any news about this PR?

**Comment by sporkmonger** [comment]
I've started working on it.

**Comment by tustvold** [comment]
Closed by https://github.com/influxdata/pbjson/pull/103

**Pull request closed without merging**